internal/pkg/manifest/backend_svc.go

```diff
@@ -199,21 +199,20 @@ func newDefaultBackendService() *BackendService {
 
 // ExposedPorts returns all the ports that are container ports available to receive traffic.
 func (b *BackendService) ExposedPorts() (ExposedPortsIndex, error) {
-	var exposedPorts []ExposedPort
+	exposedPorts := make(map[uint16]ExposedPort)
```

Comment:
> Conclusion. Modifying incoming parameters inside a method is a bad code practice you should always avoid.

Maybe we can return an updated map for each `exposePorts`? And then use the output as a parameter for the next `exposePorts`.
```

Reply:
This can be done, I was just mimicking the way the exposed ports were validated and populated. Because the two should follow the same logic, the implementation being the same makes it easier to understand, so I may do the same for the validation function.